pdg/src/graph.rs

```diff
@@ -15,47 +15,85 @@ use crate::util::ShortOption;
+    /// [`Field`] projection.
+    ///
+    /// Used for operations like `_2 = &(*_1).0`.
+    /// Nested field accesses like `_4 = &(*_1).x.y.z`
+    /// are broken into multiple [`Node`]s, each covering one level.
     Field(Field),
-    /// Pointer arithmetic.  The `isize` is the concrete offset distance.  We use this to detect
-    /// when two pointers always refer to different indices.
+
+    /// Pointer arithmetic.
+    ///
+    /// The [`isize`] is the concrete offset distance.
+    /// We use this to detect when two pointers always refer to different indices.
     Offset(isize),
 
     // Operations that can't have a `source`.
-    /// Get the address of a local.  For address-taken locals, the root node is an `AddrOfLocal`
-    /// attributed to the first statement of the function.  Taking the address of the local, as in
-    /// `_2 = &_1`, appears as a copy of that root pointer, and reading or writing from the local
-    /// shows up as a `LoadAddr` or `StoreAddr`.  This allows us to track uses of the local that
-    /// interfere with an existing reference, even when those uses don't go through a pointer.
+    /// Get the address of a [`Local`].
+    ///
+    /// For address-taken [`Local`]s, the root node is an [`AddrOfLocal`](Self::AddrOfLocal)
+    /// attributed to the first [`Statement`](rustc_middle::mir::Statement) of the function.
+    /// Taking the address of the [`Local`], as in `_2 = &_1`,
+    /// appears as a copy of that root pointer,
+    /// and reading or writing from the [`Local`]
+    /// shows up as a [`LoadAddr`](Self::LoadAddr) or [`StoreAddr`](Self::StoreAddr).
+    /// This allows us to track uses of the [`Local`]
+    /// that interfere with an existing reference,
+    /// even when those uses don't go through a pointer.
     AddrOfLocal(Local),
-    /// Get the address of a static.  These are treated the same as locals, with an
-    /// `AddressOfStatic` attributed to the first statement.
+
+    /// Get the address of a static.
+    ///
+    /// These are treated the same as [`Local`]s,
+    /// with an [`_AddressOfStatic`](Self::_AddrOfStatic) attributed to
+    /// the first [`Statement`](rustc_middle::mir::Statement).
     _AddrOfStatic(DefPathHash),
-    /// Heap allocation.  The `usize` is the number of array elements allocated; for allocations of
-    /// a single object, this value is 1.
+
+    /// Heap allocation.
+    ///
+    /// The [`usize`] is the number of array elements allocated
+    /// For allocations of a single object, this value is 1.
     Alloc(usize),
-    /// Int to pointer conversion.  Details TBD.
+
+    /// Int to pointer conversion.
+    ///
+    /// Details TBD.
     IntToPtr,
-    /// The result of loading a value through some other pointer.  Details TBD.
+
+    /// The result of loading a value through some other pointer.
+    ///
+    /// Details TBD.
     LoadValue,
 
     // Operations that can't be the `source` of any other operation.
```

Comment:
What's this referring to?

Reply:
Like above, refers to `Node::source`, which can never point to a `Node` (via `Node::dest`) which has any of the following `NodeKind`s

Reply:
Fixed in https://github.com/immunant/c2rust/pull/658/commits/dc29cfaabeb0dbc8764247e84decdad82ad3f3e8.